crates/sequencer/src/state.rs

```diff
@@ -22,3 +28,229 @@ pub struct State {
     pub storage: FxHashMap<ContractStorageKey, StarkFelt>,
     pub nonces: FxHashMap<ContractAddress, Nonce>,
 }
+
+impl BlockifierState for State {
+    fn set_storage_at(
+        &mut self,
+        contract_address: ContractAddress,
+        key: StorageKey,
+        value: StarkFelt,
+    ) {
+        self.storage.insert((contract_address, key), value);
+    }
+
+    fn increment_nonce(&mut self, contract_address: ContractAddress) -> StateResult<()> {
+        let current_nonce = self
+            .nonces
+            .get(&contract_address)
+            .cloned()
+            .unwrap_or_default();
+
+        let mut current_nonce: u64 = current_nonce.0.try_into()?;
+        current_nonce += 1;
+
+        self.nonces
+            .insert(contract_address, Nonce(StarkFelt::from(current_nonce)));
+
+        Ok(())
+    }
+
+    fn set_class_hash_at(
+        &mut self,
+        contract_address: ContractAddress,
+        class_hash: ClassHash,
+    ) -> StateResult<()> {
+        self.contracts.insert(contract_address, class_hash);
+        Ok(())
+    }
+
+    fn set_contract_class(
+        &mut self,
+        class_hash: &ClassHash,
+        contract_class: ContractClass,
+    ) -> StateResult<()> {
+        self.classes.insert(class_hash.to_owned(), contract_class);
+        Ok(())
+    }
+
+    fn set_compiled_class_hash(
+        &mut self,
+        class_hash: ClassHash,
+        compiled_class_hash: CompiledClassHash,
+    ) -> StateResult<()> {
+        self.compiled_classes
+            .insert(class_hash, compiled_class_hash);
+        Ok(())
+    }
+
+    fn to_state_diff(&self) -> CommitmentStateDiff {
+        unreachable!("to_state_diff should not be called in the sequencer")
+    }
+}
+
+impl BlockifierStateReader for State {
+    /// Default: 0 for an uninitialized contract address.
+    fn get_storage_at(
+        &mut self,
+        contract_address: ContractAddress,
+        key: StorageKey,
+    ) -> StateResult<StarkFelt> {
+        Ok(self
+            .storage
+            .get(&(contract_address, key))
+            .cloned()
+            .unwrap_or_default())
+    }
+
+    /// Default: 0 for an uninitialized contract address.
+    fn get_nonce_at(&mut self, contract_address: ContractAddress) -> StateResult<Nonce> {
+        Ok(self
+            .nonces
+            .get(&contract_address)
+            .cloned()
+            .unwrap_or_default())
+    }
+
+    /// Default: 0 (uninitialized class hash) for an uninitialized contract address.
+    fn get_class_hash_at(&mut self, contract_address: ContractAddress) -> StateResult<ClassHash> {
+        Ok(self
+            .contracts
+            .get(&contract_address)
+            .cloned()
+            .unwrap_or_default())
+    }
+
+    /// Errors if the compiled class hash is not declared.
```

Comment:
you interverted comments i think

Reply:
or at least i'm not sure why the fn is called `get_compiled_contract_class` and we access self.classes

Reply:
I think blockifier has this a little reverted, but no choice here again